### PR TITLE
Command Palette - use same focus overlay as modals, override monaco keybinding for prometheus

### LINKED ISSUE #48088: Command Palette does not automatically get focus with a query field
**What happened**: 
- In Explore, pull up Loki. Have your cursor in the query field
- Trigger the command palette with command-K. Cursor stays in query field. You can then type in the blurred out background.

**What you expected to happen**:
Command palette should get focus when triggered.

**Environment**:
- Grafana version: 9

### PR DESCRIPTION
**Pull request by gelicia**
**What this PR does / why we need it**:
We noticed a few editors had issues with the command palette. Some would keep focus when it was pulled up, so you could continue to type in the background. Others wouldn't bubble up the keypress so the command palette wouldn't display.

**Which issue(s) this PR fixes**:

Fixes #48088

**Special notes for your reviewer**: Reviewing will be best done listening to 'Deorro - Focus feat. Lena Leon' because dealing with this problem has had this song stuck in my head the last few days.

### CODE REVIEW

File: public/app/plugins/datasource/prometheus/components/monaco-query-field/MonacoQueryField.tsx [hunk 1]
```diff
@@ -199,6 +202,10 @@ const MonacoQueryField = (props: Props) => {
           editor.addCommand(monaco.KeyMod.Shift | monaco.KeyCode.Enter, () => {
             onRunQueryRef.current(editor.getValue());
           });
+
+          editor.addCommand(monaco.KeyMod.CtrlCmd | monaco.KeyCode.KeyK, function () {
```
Comment (gelicia) [inline public/app/plugins/datasource/prometheus/components/monaco-query-field/MonacoQueryField.tsx:206]: @Elfo404 🤷 😭

Comment (gelicia) [reply public/app/plugins/datasource/prometheus/components/monaco-query-field/MonacoQueryField.tsx:206]: Test with feature flag off

Comment (gelicia) [reply public/app/plugins/datasource/prometheus/components/monaco-query-field/MonacoQueryField.tsx:206]: Test with feature flag off

Comment (gelicia) [reply public/app/plugins/datasource/prometheus/components/monaco-query-field/MonacoQueryField.tsx:206]: check to see if our keybinding service sees the K part
emit the fake event and see if we can get existing toggles to pick it up

Comment (gelicia) [reply public/app/plugins/datasource/prometheus/components/monaco-query-field/MonacoQueryField.tsx:206]: check to see if our keybinding service sees the K part
emit the fake event and see if we can get existing toggles to pick it up

**Comment by gabor** [comment]
@ifrost i think the prometheus changes are fine 👍 , but i cannot comment on the `CommandPalette.tsx` changes.
(also, would be nice to get an approve from @grafana/observability-metrics , just to be sure).

**Review by ifrost — APPROVED** [review]

**Review by gabor — APPROVED** [review]
LGTM!

**Pull request merged**